WordPress/gutenberg
# Enable Block Renaming support for (almost) all blocks

## What?


Following on from https://github.com/WordPress/gutenberg/pull/53735, this PR enables the ability to rename **all blocks** by _default_, with the exception of the following blocks:

- `core/block`
- `core/template-part`
- `core/pattern`
- `core/navigation`



## Why?


Because it was suggested we should open up this feature.

## How?


- Adds metadata to all blocks by default (the `__experimentalMetadata` support is removed).
- Add new `__experimentalBlockRenaming` block supports.
- Set excluded blocks support to `false`.
- Provides custom label definitions for those blocks that already implement one.

## Testing Instructions





- Add any block (except for the exceptions defined above).
- Open list view
- Click ellipsis menu and click `Rename`.
- Enter a name.
- See block uses custom name.
- Repeat with all of the exception blocks and verify they cannot be renamed.


### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/444434/8e8dac7a-5f1f-4678-93ad-d5d81821849d
